pkg/scm/gitlab/client_actioner_assign.go

```diff
@@ -0,0 +1,102 @@
+package gitlab
+
+import (
+	"context"
+	"log/slog"
+	"math/rand"
+
+	"github.com/jippi/scm-engine/pkg/scm"
+	"github.com/jippi/scm-engine/pkg/state"
+	slogctx "github.com/veqryn/slog-context"
+)
+
+var randSource *rand.Rand
+
+func (c *Client) AssignReviewers(ctx context.Context, evalContext scm.EvalContext, update *scm.UpdateMergeRequestOptions, step scm.ActionStep) error {
+	// ensure random seed is set
+	if randSource == nil {
+		randSource = rand.New(rand.NewSource(state.RandomSeed(ctx))) //nolint:gosec
+	}
```

Comment:
I think we should use `state.RandomSeed()` here instead; now that its attachable to the context, that should be the consistent place to grab it across the code base